src/librustc_codegen_ssa/mir/analyze.rs

```diff
@@ -14,10 +14,10 @@ use super::FunctionCx;
 use crate::traits::*;
 
 pub fn non_ssa_locals<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
-    fx: &FunctionCx<'a, 'tcx, Bx>,
+    fx: &mut FunctionCx<'a, 'tcx, Bx>,
 ) -> BitSet<mir::Local> {
-    let mir = fx.mir;
-    let mut analyzer = LocalAnalyzer::new(fx);
+    let mir = fx.mir.take().unwrap();
```

Comment:
If we didn't take the `&mut BodyCache` out here, then this would fail to compile below in the usages of `fx` in the loop below. `LocalAnalyzer::new` would maintain an `&mut` borrow over `fx`.

Reply:
This way, `LocalAnalyzer` can keep a `&FunctionCx` for its lifetime, while also borrowing `&mut mir` for the `dominators` calculation.